lightning/src/ln/peer_handler.rs

```diff
@@ -345,7 +343,12 @@ pub struct PeerManager<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: De
 		RM::Target: RoutingMessageHandler,
 		L::Target: Logger {
 	message_handler: MessageHandler<CM, RM>,
-	peers: Mutex<PeerHolder<Descriptor>>,
+	peers: RwLock<PeerHolder<Descriptor>>,
+	/// Only add to this set when noise completes.
+	/// Locked *after* peers. When an item is removed, it must be removed with the `peers` write
+	/// lock held. Entries may be added with only the `peers` read lock held (though the
+	/// `Descriptor` value must already exist in `peers`).
```

Comment:
Could add a comment about what this structure is holding "Map of node pubkeys to their connection handler. One node cannot have more than one descriptor". 

Also inter-related lock handling with `peers` could be `debug_assert()" while relying on `try_write().is_err()` to observe `RwLock` is effectively held